nodejs/eks/cni.ts

```diff
@@ -100,37 +97,20 @@ function computeVpcCniYaml(yamlPath: string, args: VpcCniInputs): string {
         env.push({name: "WARM_IP_TARGET", value: args.warmIpTarget.toString()});
     }
     // Return the computed YAML.
-    return cniYaml.map(o => `---\n${jsyaml.safeDump(o)}`).join("");
+    return daemonSet;
 }
 
 function applyVpcCniYaml(yamlPath: string, args: VpcCniInputs) {
-    // Dump the kubeconfig to a file.
-    const tmpKubeconfig = tmp.fileSync();
-    fs.writeFileSync(tmpKubeconfig.fd, args.kubeconfig);
-
-    // Compute the required CNI YAML and dump it to a file.
-    const tmpYaml = tmp.fileSync();
-    fs.writeFileSync(tmpYaml.fd, computeVpcCniYaml(yamlPath, args));
-
-    // Call kubectl to apply the YAML.
-    childProcess.execSync(`kubectl apply -f ${tmpYaml.name}`, {
-        env: { ...process.env, "KUBECONFIG": tmpKubeconfig.name },
-    });
+    const daemonSet = computeVpcCniYaml(yamlPath, args);
+    const replaced = pulumi.runtime.invoke("kubernetes:kubernetes:kubectlReplace", daemonSet);
 }
 
 /**
  * VpcCni manages the configuration of the Amazon VPC CNI plugin for Kubernetes by applying its YAML chart. Once Pulumi is
- * able to programatically manage existing infrastructure, we can replace this with a real k8s resource.
+ * able to programmatically manage existing infrastructure, we can replace this with a real k8s resource.
  */
 export class VpcCni extends pulumi.dynamic.Resource {
```

Comment:
It's a little bit of a shame - I had been sort of hoping we could get rid of this whole dynamic provider.  But I think given what we implemented so far with `kubernetes:kubernetes:kubectlReplace` that is not possible.  It's interesting to consider whether this scenario (and other related ones we and customers have hit) really are suggestive of the need for a different feature in `@pulumi/kubernetes` than the one we implemented.

Of couse - this shouldn't stop us from making the progress in this PR.